backend/src/packages/chaiNNer_pytorch/pytorch/processing/image_align_rife.py

```diff
@@ -0,0 +1,211 @@
+    zip_inner_path = 'train_log'
+    download_model(download_url, model_path, model_file, zip_inner_path)
+    
+    source_h, source_w, _ = get_h_w_c(source_img)
+    target_h, target_w, _ = get_h_w_c(target_img)
+
+    #resize, then shift reference left because rife shifts slightly to the right)
+    target_img_resized = resize(target_img, (source_w, source_h), filter=ResizeFilter.LANCZOS)
+    target_img_resized = np.roll(target_img_resized, -1, axis=1)
+    target_img_resized[:, -1] = target_img_resized[:, -2]
+
+    #resize, then shift reference left because rife shifts slightly to the right)
+    #different approach to do subpixel shift
+    #scale_x = source_w / target_w
+    #scale_y = source_h / target_h
+    #shift_x = -1.0
+    #shift_y = 0
+    #transformation_matrix = np.float32([[scale_x, 0, shift_x], [0, scale_y, shift_y]])
+    #target_img_resized = cv2.warpAffine(target_img, transformation_matrix, (source_w, source_h), flags=cv2.INTER_LANCZOS4, borderMode=cv2.BORDER_REPLICATE)
+    
+    #padding because rife can only work with multiples of 32 (changes with precision mode)
+    pad_h, pad_w = calculate_padding(source_h, source_w, precision_mode)
+    top_pad = pad_h // 2
+    bottom_pad = pad_h - top_pad
+    left_pad = pad_w // 2
+    right_pad = pad_w - left_pad
+    target_img_padded = np.pad(target_img_resized, ((top_pad, bottom_pad), (left_pad, right_pad), (0, 0)), mode='edge')
+    source_img_padded = np.pad(source_img, ((top_pad, bottom_pad), (left_pad, right_pad), (0, 0)), mode='edge')
+    #target_img_padded = target_img
+    #source_img_padded = source_img_resized
+
+    exec_options = get_settings(context)
+    device = exec_options.device
+
+    #load model
+    model_full_path = os.path.join(model_path, model_file)
+    model = IFNet().to(device)#.half()
+    state_dict = torch.load(model_full_path, map_location=device)
+    new_state_dict = {k.replace('module.', ''): v for k, v in state_dict.items()}
+    model.load_state_dict(new_state_dict)
+    model.eval()
+
+    #convert to tensors
+    target_tensor_padded = np2tensor(target_img_padded, change_range=True).to(device)#.half()
+    source_tensor_padded = np2tensor(source_img_padded, change_range=True).to(device)#.half()
+
+    #concatenate images
+    img_pair = torch.cat((target_tensor_padded, source_tensor_padded), dim=1)
+
+    with torch.no_grad():
+        aligned_img, _ = model(img_pair, multiplier=multiplier, num_iterations=alignment_passes, blur_strength=blur_strength, ensemble=ensemble, device=device)
+
+    #convert back to numpy and crop
+    result_img = tensor2np(aligned_img.squeeze(0).cpu(), change_range=False, imtype=np.float32)
+    result_img = result_img[top_pad:top_pad+source_h, left_pad:left_pad+source_w]
+
+    return result_img
+
+
+
+
+
+@processing_group.register(
+    schema_id="chainner:pytorch:image_align_rife",
+    name="Image Align with Rife",

```

Comment:
do you mind renaming this to "Align Image to Reference"

Reply:
No problem. My thinking with mentioning Rife in the name was in case in the future different kinds of alignment methods are added. For example it may make sense to add a traditional affine alignment method at some point, which would be better with images that are off by a lot, which could then be refined by this method. But maybe it makes more sense to think about that if it happens. What do you think?

Reply:
Usually the way we handle this is we have the node named after the action and we have tabs or a dropdown for the implementation and it's options. There's no need to have multiple nodes for all the different ways to align

Reply:
Okay! Should we combine the average color fix and the wavelet color fix into one node as well?